c++/src/kj/compat/http.c++

```diff
@@ -4956,9 +5136,12 @@ public:
 
     auto id = ++counter;
 
-    return httpInput.readResponseHeaders().then(
-        [this,id](HttpHeaders::ResponseOrProtocolError&& responseOrProtocolError)
-            -> HttpClient::ConnectResponse {
+    auto paf = kj::newPromiseAndFulfiller<kj::Maybe<HttpInputStreamImpl::ReleasedBuffer>>();
```

Comment:
Yeah `split()` is pretty much the only place where I willingly use `Tuple`...

Reply:
Updated, but can you verify the use here. Specifically, I'm returning exceptions in two cases as part of the tuple to cause the read guard to be rejected and want to be certain that's the correct approach using tuple and split.

Reply:
Yeah I guess that works. It's a little funny that you end up returning a `Promise` that always ends up being immediately fulfilled but I guess that's the only way to make one branch of the split error without erroring the other branch.

Reply:
Yeah, that's precisely the kind of weirdness I was trying to avoid by eliminating the split but ah well... we'll go with this unless you happen to think of a better suggestion